views/js/glogin.js

```diff
@@ -0,0 +1,24 @@
+var googleUser = {};
+var startApp = function (button, name) {
+
+  gapi.load('auth2', function () {
+
+    // Retrieve the singleton for the GoogleAuth library and set up the client.
+
+    auth2 = gapi.auth2.init({
+      client_id: '594743792021-tvffn11n961cqea8mbufm9gnts372m0n.apps.googleusercontent.com',
+      cookiepolicy: 'single_host_origin',
+    });
+
+    console.log(button.id);
+
+    auth2.attachClickHandler(button, {},
+      function (googleUser) {
+        name.innerText = "Signed in: " +
+          googleUser.getBasicProfile().getName();
+      }, function (error) {
+        alert(JSON.stringify(error, undefined, 2));
```

Comment:
As in GoogleAPi Call we specify our server IP and if the GoogleApi Token Do not matched with the Ip on which server is running an error will be displayed over here.
For them you can use bootstrap alert box instead of simple browser alert window (https://getbootstrap.com/docs/4.0/components/alerts/)
And Please try to analyze what all errors you get. For example If internet is not working, If the Ip and Client Token Do no Match, etc. To analyze what kind of error message is google returning in this case. so instead of Directly doing JSON.stringify() we can use a message/Description in the JSON to display instead of displaying whole error object.

Reply:
Resolving As moved to Passport.